pkg/kubelet/kubelet_test.go

```diff
@@ -2154,6 +2156,65 @@ func TestMakeEnvironmentVariables(t *testing.T) {
 				"KUBERNETES_RO_PORT_8087_TCP_ADDR=1.2.3.7"),
 			21,
 		},
+		{
+			"simple substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
+					{Name: "FOO", Value: "${BAR}"},
+				},
+			},
+			"kubernetes",
+			true,
+			util.NewStringSet(
+				"BAR=bar",
+				"FOO=bar",
+			),
+			2,
+		},
+		{
+			"slightly more complex substitution",
+			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name: "test-pod"}},
+			&api.Container{
+				Env: []api.EnvVar{
+					{Name: "BAR", Value: "bar"},
```

Comment:
It was for specific fields that are not cleanly represented by an API resource.  I think ObjectReference makes sense for many cases, but not all data we expose necessarily has an API object (although maybe you could argue that they should).

----- Original Message -----
> > +				},
> > +			},
> > +			"kubernetes",
> > +			true,
> > +			util.NewStringSet(
> > +				"BAR=bar",
> > +				"FOO=bar",
> > +			),
> > +			2,
> > +		},
> > +		{
> > +			"slightly more complex substitution",
> > +			&api.Pod{ObjectMeta: api.ObjectMeta{Namespace: "unknown", Name:
> > "test-pod"}},
> > +			&api.Container{
> > +				Env: []api.EnvVar{
> > +					{Name: "BAR", Value: "bar"},
> 
> @smarterclayton On second reading, I don't get the bools for fields thing.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6000/files#r27497979